include/veriblock/entities/btcblock.hpp

```diff
@@ -20,8 +21,10 @@ struct BtcBlock {
   static BtcBlock fromRaw(ReadStream& stream) {
     BtcBlock block{};
     block.version = stream.readLE<uint32_t>();
-    block.previousBlock = stream.readSlice(SHA256_HASH_SIZE);
-    block.merkleRoot = stream.readSlice(SHA256_HASH_SIZE);
+    block.previousBlock =
+        ((Sha256Hash)stream.readSlice(SHA256_HASH_SIZE)).reverse();
+    block.merkleRoot =
```

Comment:
Semantically it is wrong to cast a slice to Sha256Hash.
I think better would be to add method `std::vector<uint8_t> reverse() const noexcept` to a Slice